client/packages/system/src/Item/api/hooks/useItems.ts

```diff
@@ -46,11 +46,13 @@ export const useVisibleOrOnHandItems = (queryParams: ItemParams) => {
     };
 
     // if using the hasStockOnHand filter, replace the isVisibleOrOnHand filter with hasStockOnHand
-    if (filterBy?.['hasStockOnHand'] !== undefined) {
+    let hasStockOnHand = filterBy?.['hasStockOnHand'];
+    if (hasStockOnHand !== undefined) {
       filter = {
         ...filter,
         isVisibleOrOnHand: undefined,
-        hasStockOnHand: !!filterBy?.['hasStockOnHand'],
+        hasStockOnHand: !!hasStockOnHand,
+        isVisible: !hasStockOnHand ? true : undefined,
```

Comment:
Hm, this is starting to seem a bit unwieldy to me... what do you think of extracting like this?

https://github.com/msupply-foundation/open-msupply/compare/9302-align-dashboard-numbers...9302-align-dashboard-numbers(on-hand-filter-suggestion)